lib/spack/docs/configuration.rst

```diff
@@ -4,58 +4,136 @@
+
+   module_roots:
+     tcl:    $spack/share/spack/modules
+     lmod:   $spack/share/spack/lmod
+     dotkit: $spack/share/spack/dotkit
+
+^^^^^^^^^^^^^^^^^^^
+``build_stage``
+^^^^^^^^^^^^^^^^^^^
+
+Spack is designed to run out of a user home directories, and on many
+systems the home directory is NFS mounted *not* a fast filesystem.  We
+create build stages in a temporary directory to avoid this.  Many systems
+impose quotas on home directories, and ``/tmp`` or similar directories
+often have more available space.
+
+By default, the ``build_stage`` is set like this:
+
+.. code-block:: yaml
+
+   build_stage:
+    - $tempdir
+    - /nfs/tmp2/$user
+    - $spack/var/spack/stage
+
+This is an ordered list of paths that Spack should search when trying to
+find a temporary directory for the build stage.  The list is searched in
+order, and Spack will use the first directory to which it has write access.
+
+When Spack creates a temporary build stage, it places a symbolic link to
+the stage in ``$spack/var/spack/stage`` so that it can track the stage
+and clean it up later.  By default, build stages are removed after a
+successful package build.  You can manually clean build stages with
+:ref:`spack purge <cmd-spack-purge>`.
+
+Note that the last item in the list is ``$spack/var/spack/stage``.  If
+this is used as the build stage, Spack will build *directly* in its
+prefix and will not use temporary space.
+
+^^^^^^^^^^^^^^^^^^^^
+``source_cache``
+^^^^^^^^^^^^^^^^^^^^
+
+Location to cache downloaded tarballs and repositories.  By default these
+are stored in ``$spack/var/spack/cache``.  These are stored indefinitely
+by default. See :ref:`spack purge <cmd-spack-purge>`.
+
+^^^^^^^^^^^^^^^^^^^^
+``misc_cache``
+^^^^^^^^^^^^^^^^^^^^
+
+Temporary directory to store long-lived cache files, such as indices of
+packages available in repositories.  Defaults to ``~/.spack/cache``.
+
+
+^^^^^^^^^^^^^^^^^^^^
+``verify_ssl``
+^^^^^^^^^^^^^^^^^^^^
+
+When set to ``true`` (default) Spack will verify certificates of remote
+hosts when making ``ssl`` connections.  Set to ``false`` to disable, and
+tools like ``curl`` will use their ``--insecure`` options.  Disabling
+this can expose you to attacks.  Use at your own risk.
+
+^^^^^^^^^^^^^^^^^^^^
+``checksum``
+^^^^^^^^^^^^^^^^^^^^
+
+When set to ``true``, Spack verifies downloaded source code using a
+checksum, and will refuse to build packages that it cannot verify.  Set
+to ``false`` to disable these checks.  Disabling this can expose you to
+attacks.  Use at your own risk.
+
+^^^^^^^^^^^^^^^^^^^^
+``clean``
+^^^^^^^^^^^^^^^^^^^^
+
+By default, Spack clears variables in your environment that can change
+the way packages build. This includes ``LD_LIBRARY_PATH``, ``CPATH``,
+``LIBRARY_PATH``, ``DYLD_LIBRARY_PATH``, and others.
+
+You can set this to ``false`` to skip the environmnet cleaning step and
+build in a "dirty" environment.  Be aware that this will reduce the
+reproducibility of builds.
+
```

Comment:
Again... explain cases where `clean=False` is necessary.

Also... are `True` and `true` both allowed in the config file format?  Pythonistas are all trained to write `True` and `False`; forcing users to do `true` and `false` would introduce a useless distinction.